### PR TITLE
[euphoria-core] #177 added MultiDataSink

### PR DESCRIPTION
**Pull request by mareksimunek**

### CODE REVIEW

File: euphoria-core/src/main/java/cz/seznam/euphoria/core/client/io/MultiDataSink.java [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+  }
+
+  /**
+   * Selects DataSink where to save output elements
+   * @param splitFunction transform
+   * @param <SPLIT> key to select DataSink
+   * @param <IN> type of input element
+   * @return Builder
+   */
+  public static <SPLIT, IN> Builder<SPLIT, IN> selectBy(UnaryFunction<IN, SPLIT> splitFunction) {
```
Comment (dmvk) [inline euphoria-core/src/main/java/cz/seznam/euphoria/core/client/io/MultiDataSink.java]: how about renaming splitFunction to selectFunction?

File: euphoria-core/src/main/java/cz/seznam/euphoria/core/client/io/MultiDataSink.java [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+ *   .addSink(Type.FIRST, InputElement::getFirst, new SequenceFileSink<>(...))
+ *   .addSink(Type.SECOND, InputElement::getSecond, new StdoutSink<>(...))
+ *   .addSink(Type.THIRD, InputElement::getThird, new StdoutSink<>(...))
+ *   .build();
+ * }
+ *</pre>
+ * @param <SPLIT> key to select DataSink where saves the element
+ * @param <IN> type of input element
+ */
+public class MultiDataSink<SPLIT, IN> implements DataSink<IN> {
```
Comment (dmvk) [inline euphoria-core/src/main/java/cz/seznam/euphoria/core/client/io/MultiDataSink.java]: SPLIT generic type should have a better name.. how about SINK_ID or just ID?

File: euphoria-core/src/main/java/cz/seznam/euphoria/core/client/io/MultiDataSink.java [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+
+    public DataSink<IN> build() {
+      Map<SPLIT, DataSinkWrapper<SPLIT, IN, ?>> sinksMap = new HashMap<>();
+      dataSinkWrappers.forEach((el) -> sinksMap.put(el.getType(), el));
+      return new MultiDataSink<>(splitFunction, sinksMap);
+    }
+  }
+
+  private static class DataSinkWrapper<SPLIT, IN, OUT> {
+    private final SPLIT type;
```
Comment (dmvk) [inline euphoria-core/src/main/java/cz/seznam/euphoria/core/client/io/MultiDataSink.java]: this should be also renamed to `id`

File: euphoria-core/src/main/java/cz/seznam/euphoria/core/client/io/MultiDataSink.java (hunk 1 above)
Comment (je-ik) [reply euphoria-core/src/main/java/cz/seznam/euphoria/core/client/io/MultiDataSink.java]: :+1:

File: pom.xml [hunk 4]
```diff
@@ -540,7 +540,8 @@
       <dependency>
         <groupId>org.projectlombok</groupId>
         <artifactId>lombok</artifactId>
-        <version>1.14.8</version>
+        <version>1.16.18</version>
+        <scope>compile</scope>
```
Comment (je-ik) [inline pom.xml]: :+1:

**Comment by dmvk** [comment]
Closes #177

**Pull request merged**